presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayIntersectFunction.java

```diff
@@ -0,0 +1,154 @@
+import static com.facebook.presto.type.TypeUtils.readStructuralBlock;
+import static com.facebook.presto.type.TypeUtils.buildStructuralSlice;
+import static com.facebook.presto.type.TypeUtils.parameterizedTypeName;
+import static com.facebook.presto.util.Reflection.methodHandle;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.String.format;
+
+public final class ArrayIntersectFunction
+        extends ParametricScalar
+{
+    public static final ArrayIntersectFunction ARRAY_INTERSECT_FUNCTION = new ArrayIntersectFunction();
+    private static final String FUNCTION_NAME = "array_intersect";
+    private static final Signature SIGNATURE = new Signature(FUNCTION_NAME, ImmutableList.of(comparableTypeParameter("E")), "array<E>", ImmutableList.of("array<E>", "array<E>"), false, false);
+    private static final MethodHandle METHOD_HANDLE = methodHandle(ArrayIntersectFunction.class, "intersect", Type.class, Slice.class, Slice.class);
+
+    @Override
+    public Signature getSignature()
+    {
+        return SIGNATURE;
+    }
+
+    @Override
+    public boolean isHidden()
+    {
+        return false;
+    }
+
+    @Override
+    public boolean isDeterministic()
+    {
+        return true;
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return "Intersects elements of the two given arrays.";
+    }
+
+    @Override
+    public FunctionInfo specialize(Map<String, Type> types, int arity, TypeManager typeManager, FunctionRegistry functionRegistry)
+    {
+        checkArgument(types.size() == 1, format("%s expects only one argument", FUNCTION_NAME));
+        TypeSignature typeSignature = parameterizedTypeName("array", types.get("E").getTypeSignature());
+        MethodHandle methodHandle = METHOD_HANDLE.bindTo(types.get("E"));
+        Signature signature = new Signature(FUNCTION_NAME, typeSignature, typeSignature, typeSignature);
+        return new FunctionInfo(signature, getDescription(), isHidden(), methodHandle, isDeterministic(), false, ImmutableList.of(false, false));
+    }
+
+    public static Slice intersect(Type type, Slice leftArray, Slice rightArray)
+    {
+        Block leftBlock = readStructuralBlock(leftArray);
+        Block rightBlock = readStructuralBlock(rightArray);
+
+        int leftPositionCount = leftBlock.getPositionCount();
+        int rightPositionCount = rightBlock.getPositionCount();
+        int leftPosition = 0;
+        int rightPosition = 0;
+
+        List<Integer> leftPositions = Ints.asList(new int[leftPositionCount]);
+        List<Integer> rightPositions = Ints.asList(new int[rightPositionCount]);
+
+        for (int i = 0; i < leftPositionCount; i++) {
+            leftPositions.set(i, i);
+        }
+        for (int i = 0; i < rightPositionCount; i++) {
+            rightPositions.set(i, i);
+        }
+
+        Collections.sort(leftPositions, new Comparator<Integer>()
```

Comment:
I think it will be more efficient if you use `IntArrays.quickSort()` because then we can avoid the boxing of `Integer`